File: JS/LeetC/g75/canAttendMeetings.js
function canAttendMeetings(intervals) {
  intervals.sort(([a1, b1], [a2, b2]) => a1- a2);

  for (let i = 0; i < intervals.length - 1; i++) {
    let current = intervals[i];
    let next = intervals[i + 1];

    if (next[0] < current[1] || current[0] > next[1]) {
      return false;
    }
  }

  return true;
}

File: Java/hellointerview/03-intervals/canAttendMeetings.java
public class Solution {
  public Boolean canAttenMeetings(int[][] intervals) {
    Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
    for (int i = 0; i < intervals.length - 1; i++) {
      if (intervals[i + 1][0] < intervals[i][1]) {
        return false;
      }
    }
    return true;
  }
}
